test/fixtures_isce3.py

```diff
@@ -0,0 +1,171 @@
+
+
+@fixture(scope=determine_scope)
+def env_files_dir_path() -> Path:
+    """Return the path to the ISCE3 environment files directory."""
+    return Path(__file__).parent.parent / "env_files"
+
+
+@fixture(scope=determine_scope)
+def runtime_lockfile_path(env_files_dir_path: Path) -> Path:
+    """Return the path to the ISCE3 runtime lock file."""
+    return env_files_dir_path / "lock-runtime.txt"
+
+
+@fixture(scope=determine_scope)
+def dev_lockfile_path(env_files_dir_path: Path) -> Path:
+    """Return the path to the ISCE3 dev lock file."""
+    return env_files_dir_path / "lock-dev.txt"
+
+
+@fixture(scope=determine_scope)
+def isce3_build_tag() -> str:
+    """Return a tag for images in the ISCE3 build chain."""
+    return generate_tag("isce3-setup")
+
+
+@fixture(scope=determine_scope)
+def isce3_setup_images(
+    isce3_build_tag: str,
+    base_tag: str,
+    cuda_version: Tuple[int, int],
+    cuda_repo_ver: str,
+    runtime_lockfile_path: Path,
+    dev_lockfile_path: Path,
+) -> Iterator[dict[str, Image]]:
+    """Return a dictionary of ISCE3 setup images."""
+    cuda_major, cuda_minor = cuda_version
+
+    image_dict = setup_all(
+        base=base_tag,
+        tag=isce3_build_tag,
+        no_cache=False,
+        cuda_version=f"{cuda_major}.{cuda_minor}",
+        cuda_repo=cuda_repo_ver,
+        runtime_env_file=runtime_lockfile_path,
+        dev_env_file=dev_lockfile_path,
+        verbose=True,
+        test=True,
+    )
+
+    yield image_dict
+
+    for tag in image_dict:
+        remove_docker_image(tag)
+
+
+@fixture(scope=determine_scope)
+def isce3_env_dev_image_tag(
+    isce3_build_tag: str,
+    isce3_setup_images: dict[str, Image],
+) -> str:
+    """Return the tag of the ISCE3 development environment image."""
+    pattern = re.compile(rf".*{isce3_build_tag}.*mamba-dev")
+
+    env_dev_image_tag: str = ""
+
+    for tag in isce3_setup_images:
+        if pattern.match(tag):
+            env_dev_image_tag = tag
+
+    if not env_dev_image_tag:
+        raise ValueError("No development environment image tag found.")
+
+    return env_dev_image_tag
+
+
+@fixture(scope=determine_scope)
+def isce3_git_repo_tag() -> str:
+    """Return a tag for the ISCE3 repository image."""
+    return generate_tag("isce3-repo")
+
+
+@fixture(scope=determine_scope)
+def isce3_git_repo_image(
+    isce3_env_dev_image_tag: str,
+    isce3_git_repo_tag: str,
+    base_properties: Tuple[Any, URLReader],
+) -> Iterator[Image]:
+    """Return a tag for the ISCE3 repository image."""
+    archive = "https://github.com/isce-framework/isce3/archive/refs/tags/v0.16.0.tar.gz"
+    _, url_reader = base_properties
+
+    dockerfile = git_extract_dockerfile(
+        base=isce3_env_dev_image_tag,
+        archive_url=archive,
+        directory=Path("/src/"),
+        url_reader=url_reader,
+    )
+
+    yield Image.build(
+        tag=isce3_git_repo_tag, dockerfile_string=dockerfile, no_cache=True
```

Comment:
Are we deliberately using `no_cache=True` here and for the config/build images below? I thought based on our slack convo the other day that we were OK with using the cache for these images.